### PR TITLE
Add a student feature

### PR DESCRIPTION
**Pull request by ankurnath**
Added the feature and the tests

### CODE REVIEW

File: zlp-scheduler/app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -220,4 +220,34 @@ def student_actions
     @actions = StudentAction.all.order(created_at: :desc)
   end
   
+  def add_student
```
Comment (henrywang1) [inline zlp-scheduler/app/controllers/admin_controller.rb:223]: You can add `add_student` to `before_action :require_admin` in the beginning of the file.

File: zlp-scheduler/app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -220,4 +220,34 @@ def student_actions
...
+        if User.find_by(email:params[:user][:email]).nil?
+          @user = User.new()
+          @user.role = 'student'
+          @user.firstname=params[:user][:firstname]
+          
+          @user.lastname=params[:user][:lastname]
+          @user.email=params[:user][:email]
+          @user.cohort_id = session[:cohort_id]
+          @user.uin = params[:user][:uin]
+          @user.password='Temp'
```
Comment (henrywang1) [inline zlp-scheduler/app/controllers/admin_controller.rb:241]: You can set `@user.activate = false`, so new students cannot use 'Temp' to login before claim account.

File: zlp-scheduler/app/views/admin/add_student.html.haml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+%div#addAdminDiv  
+    %h1{:style => "text-align:center"} Add a new Student to #{@cohort_name}
+    = form_for @user,url: add_student_path, method: "post" do |f| 
+        %table#userTable
+            %tr
+                %th#specialTableHeader First Name
+                %th#specialTableHeader Last Name
+                %th#specialTableHeader TAMU Email
```
Comment (henrywang1) [inline zlp-scheduler/app/views/admin/add_student.html.haml:8]: You can set both email and UIN to `required => true`

Comment (ankurnath) [reply zlp-scheduler/app/views/admin/add_student.html.haml:8]: Fixed

File: zlp-scheduler/app/controllers/admin_controller.rb (hunk 2 above)
Comment (ankurnath) [reply zlp-scheduler/app/controllers/admin_controller.rb:241]: Fixed

**Review by henrywang1 — APPROVED** [review]

**Pull request merged**